### PR TITLE
feat: migrate keycloak management from prodpublick8s to publick8s

### PR DESCRIPTION
**Pull request by dduportal**
Related to https://github.com/jenkins-infra/helpdesk/issues/3351, this PR is the first step to the migration of Keycloak to the new cluster `publick8s`.

It introduces the following changes:

- Stop managing keycloak in `prodpublick8s`
- Add the keycloak "admin" backend only in `publick8s` 

Please note that the resulting instance will fail at first:

- The database configuration is not updated yet (still points at the old version)
- No "public" access (the ingress `beta.accounts.jenkins.io` created by the jenkins-infra/keycloak-public release was removed 2 years ago)
- LDAP should allow requests from the new keycloak but to be verified

### CODE REVIEW

File: clusters/publick8s.yaml [hunk 1]
```diff
@@ -111,3 +114,13 @@ releases:
...
       - "../config/wiki.yaml"
+  - name: keycloak # Admin: https://admin.accounts.jenkins.io/auth/admin
+    namespace: keycloak
+    chart: codecentric/keycloak
+    version: 18.4.3
+    timeout: 600
+    values:
+      - "../config/keycloak.yaml"
+    secrets:
+      - "../secrets/config/keycloak/db.yaml"
```
[the remark this answers is not in the document]
Comment (lemeurherve) [reply clusters/publick8s.yaml:125]:
```suggestion
      - "../secrets/config/keycloak/public-db.yaml"
```

[the remark this answers is not in the document]
Comment (dduportal) [reply clusters/publick8s.yaml:125]: good point, let me do it

File: clusters/prodpublick8s.yaml [hunk 2]
```diff
@@ -146,23 +146,6 @@ releases:
...
-    secrets:
-      - "../secrets/config/keycloak/db.yaml"
-      - "../secrets/config/keycloak/http.yaml"
-  - name: keycloak-public # User: https://admin.accounts.jenkins.io/auth/realms/jenkins/account
-    namespace: keycloak
-    chart: jenkins-infra/keycloak
-    version: 0.1.1
-    timeout: 150
-    values:
-      - "../config/keycloak-public.yaml"
```
Comment (lemeurherve) [inline clusters/prodpublick8s.yaml:165]: Maybe remove keycloak-public in a separate PR?

File: clusters/prodpublick8s.yaml
Comment (lemeurherve) [inline clusters/prodpublick8s.yaml:1]: The repositories section should be updated here too by removing L7 to L9

File: clusters/publick8s.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dduportal) [reply clusters/publick8s.yaml]: Done (and added the new secret in https://github.com/jenkins-infra/charts-secrets/commit/9adca8ec9b4c925ca67c51d1377860dad5a96005)

File: clusters/prodpublick8s.yaml (hunk 2 above)
Comment (dduportal) [reply clusters/prodpublick8s.yaml:165]: I opeted for the same pattern as in https://github.com/jenkins-infra/kubernetes-management/pull/3945. We can split in 2 not a problem

Comment (lemeurherve) [reply clusters/prodpublick8s.yaml]: The deletion of keycloak-public is distinct from the migration of keycloak (admin), not exactly the same purpose.

As you like 🙂

Comment (lemeurherve) [reply clusters/prodpublick8s.yaml]: I would keep the deletion of the keycloak (admin) release from clusters/prodpublick8s.yaml in this PR in any case.

Comment (dduportal) [reply clusters/prodpublick8s.yaml]: Oh I see, another incoming update.

A 2nd PR only for removing the "public" would make sense for auditability, while "single" PR make sense for "migration" so both are OK for me.

File: clusters/prodpublick8s.yaml
Comment (dduportal) [reply clusters/prodpublick8s.yaml:1]: Done

File: clusters/prodpublick8s.yaml (hunk 2 above)
Comment (dduportal) [reply clusters/prodpublick8s.yaml]: https://github.com/jenkins-infra/kubernetes-management/pull/3984 here you are

**Review by lemeurherve — APPROVED** [review]
🚀

**Pull request merged**